src/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -429,6 +429,80 @@ public static bool TrySkipBlock(BinaryReader reader)
 
             return true;
         }
+
+        public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+        {
+            const int OffsetToFilenameLength = 22; // from the point after the signature
+
+            if (reader.ReadUInt32() != SignatureConstant)
+                return false;
+
+            if (reader.BaseStream.Length < reader.BaseStream.Position + OffsetToFilenameLength)
+                return false;
+
+            uint minimunVersion = reader.ReadUInt16();
+            int dataDescriptorBit = reader.ReadUInt16() & 0x08;
+            reader.BaseStream.Seek(10, SeekOrigin.Current);
+            long compressedSize = reader.ReadUInt32();
+            long uncompressedSize = reader.ReadUInt32();
+            int filenameLength = reader.ReadUInt16();
+            uint extraFieldLength = reader.ReadUInt16();
+            reader.BaseStream.Seek(filenameLength, SeekOrigin.Current);
+            long endExtraFields = reader.BaseStream.Position + extraFieldLength;
+            
+            if (dataDescriptorBit == 0)
+            {
+                bool uncompressedSizeInZip64 = uncompressedSize == ZipHelper.Mask32Bit;
+                bool compressedSizeInZip64 = compressedSize == ZipHelper.Mask32Bit;
+                Zip64ExtraField zip64;
+
+                // Ideally we should also check if the minimunVersion is 64 bit or above, but there could zip files for which this version is not set correctly
+                if (uncompressedSizeInZip64 || compressedSizeInZip64)
+                {
+                    zip64 = Zip64ExtraField.GetJustZip64Block(new SubReadStream(reader.BaseStream, reader.BaseStream.Position, extraFieldLength), uncompressedSizeInZip64, compressedSizeInZip64, false, false);
+
+                    if (zip64.UncompressedSize != null)
+                        uncompressedSize = zip64.UncompressedSize.Value;
+                    if (zip64.CompressedSize != null)
+                        compressedSize = zip64.CompressedSize.Value;
+                }
+
+                reader.BaseStream.AdvanceToPosition(endExtraFields);
+            }
+            else
+            {
+                bool is64bit = minimunVersion > 44;
+                long seekSize = entry.CompressedLength + (is64bit ? 8 : 4) * 2;
+                uint dataDescriptorSignature = reader.ReadUInt32();
```

Comment:
I think you're reading this from the wrong location.  At this point the stream is still pointing in the local header, but the data descriptor signature should be at the end of the file.  I believe you end up skiping over the actual data descriptor signature on line 484 below (when you added + 4).